src/config.js

```diff
@@ -334,7 +335,7 @@ export default class Config {
 
     this.disablePrepublish = !!opts.disablePrepublish;
 
-    this.nonInteractive = !!opts.nonInteractive;
+    this.nonInteractive = opts.nonInteractive != null ? !!opts.nonInteractive : isCi || !process.stdout.isTTY;
```

Comment:
The though was "if `opts.nonInteractive` is set to anything, respect that. Otherwise check if we are on CI or stdout is non-tty".
Can simplify it a bit?